src/ConvexHull.jl

```diff
@@ -47,15 +49,10 @@ Return the support vector of a convex hull in a given direction.
 - `d`  -- direction
 - `ch` -- the convex hull of two sets
 """
-function σ(d::AbstractVector{Float64}, ch::ConvexHull)::Vector{Float64}
+function σ(d::AbstractVector{<:Real}, ch::ConvexHull)::Vector{<:Real}
```

Comment:
here why not `function σ(d::AbstractVector, ch::ConvexHull)` ?

Reply:
Counter question:
Why not check that the input is of real number type and the output is a) a `Vector` and b) of real number type?

Reply:
alright, we *don't check* that input/output number types are the same here, for that we would need to use `N` as well. 

let's go with `<:Real` then

Reply:
for the output type, i just saw that if we want to allow sparse vectors for input (`d`) then it should also be a `AbstractVector`.

Reply:
e.g. we want that `σ(sparsevec([1],[1.0],2), BallInf(ones(2),1.0)` works -- returning a sparsevector in some cases. maybe not in all cases

Reply:
Yes, I was also thinking about that. Sometimes we have `AbstractVector` and sometimes we have `Vector`.
The user can of course still expect an `AbstractVector`.

If the implementation ensures that the type is always a `Vector`, everything will work, but I still prefer to keep the return type more general so that a caller does not rely on that type.

On a side note, when I click on the file, then it shows me an `AbstractVector` oO

Reply:
yes. the implementation was indeed not working if you passed a sparse vector to the `BallInf` constructor (because the output type would also be an abstract vector, and in that case it didn't match the `Vector` output annotation).

Reply:
what i was saying above is that it's not true in all cases that we respect the type of the input direction. either rational -> float64 or sparse -> dense conversions may occur. this is a case by case study that we should see but can do it in separate issues.

Reply:
please if u agree [with the note just added](https://github.com/JuliaReach/LazySets.jl/pull/37/commits/5c3b4f2df57c00537286a239637d274c68b2e713). this is to say that the rational computations are currently not reliable -- the output support vector is *not* a vertex of the given set! (assuming it's exact)

Reply:
agreed

Reply:
let's use `AbstractFloat` instead of real in that case, then it just say "i don't know" if you throw it rational.